### DISCUSSION TITLE
Why is bun install faster than npm install, what is the key tech ?

### DISCUSSION THREAD

**Discussion by zhangzhuang15** [oven-sh/bun#16052] [Q&A]
I hear that bun install is faster than npm install from bun official website, but I don't find some explaining about reason in detail. I'm not Zig expert and cannot get the answer by reading source code on my own. So, could someone help me understand why bun install is faster and what the key tech it depends on ? Thank you.

**Comment by hubdidub** [comment]
Jarred wrote something about bun install here: https://bun.sh/blog/bun-lock-text-lockfile however the blog post is more about the text lockfile format and less about bun install in general. So it's not an direct answer to your question, but still might be interesting for you.

**Reply by zhangzhuang15** [reply]
Thank you @hubdidub , I have checked that blog, it helps me a lot, now I can get idea what bun install is faster than npm install, and I can't wait to learn more details from bun teamwork. @Jarred-Sumner Any thoughts about posting a blog introducing more details of making bun install faster ? As a tech fan, I'm looking forwards to this new blog. By the way, what do other people think about this, are you waiting for such a blog ?
